lib/api/src/sys/instance.rs

```diff
@@ -118,12 +113,52 @@ impl Instance {
     /// Those are, as defined by the spec:
     ///  * Link errors that happen when plugging the imports into the instance
     ///  * Runtime errors that happen when running the module `start` function.
-    pub fn new(
-        module: &Module,
-        resolver: &(dyn Resolver + Send + Sync),
-    ) -> Result<Self, InstantiationError> {
+    pub fn new(module: &Module, imports: Imports) -> Result<Self, InstantiationError> {
```

Comment:
In hindsight we probably shouldn't take `Imports` by value and take it by reference like it was before instead. There's no benefit to taking it by value and it could be reused to make more instances.

Reply:
I think we needed to be owned for the JS use case?

Reply:
Maybe we could encourage `Clone`ing in this case?

Reply:
I can't remember why we decided it would be needed for JS. I don't think the reasoning applies any more, but I could be wrong.

Reply:
Please change this back to taking `Imports` by reference.